m2plots/time-series/latestyear_spaghetti_ncaregions.py

```diff
@@ -12,12 +12,13 @@
 warnings.filterwarnings('ignore')
 import sys
 
-#Required command line inputs in order are 1) the variable to be plotted, 2) the region, 3) the year to be highlighted, and 4) the ending month of the year to be highlighted. 
-#Example usage: ./latestyear_spaghetti_ncaregions.py t2m ne 2023 12
+#Required command line inputs in order are 1) the variable to be plotted, 2) the region, 3) the year to be highlighted, 4) the ending month of the year to be highlighted, and 5) the stream to be used (NRT ops vs retro). 
+#Example usage: ./latestyear_spaghetti_ncaregions.py t2m ne 2023 12 ops
 yamlkey_var=sys.argv[1]
 yamlkey_reg=sys.argv[2]
 endyear=int(sys.argv[3])
 endmonth=int(sys.argv[4])
+yamlkey_stream=sys.argv[5]
```

Comment:
Fine for now, but for future reference, a much more user-friendly and robust approach to parsing command line arguments in Python programs is to use the builtin `argparse` library.

It's a bit more code, but produces a much more flexible interface, as well as automated help, type checking, and other goodies.

A direct translation of the current functionality --- where the arguments are all positional --- would look like this:

```python
import argparse

parser = argparse.ArgumentParser()  # Can add descriptions of arguments here, etc.
parser.add_argument('yamlkey_var')
parser.add_argument('yamlkey_reg')
parser.add_argument('endyear', type=int)
parser.add_arugment('endmonth', type=int)
parser.add_argument('yamlkey_stream')

args = parser.parse_args()
yamlkey_var = args.yamlkey_var
yamlkey_reg = args.yamlkey_reg
endyear = args.endyear
endmonth = args.endmonth
yamlkey_stream = args.yamlkey_stream
```

However, we can improve the flexibility and readability of these scripts by leveraging more features of `argparse` (such as making certain arguments optional with default values; allowing arguments to be passed in different order by labeling them; etc.)